### PR TITLE
[ELY-2548] BasicAuthenticationMechanism should return FORBIDDEN instead of UNAUTHORIZED

### PR DESCRIPTION
**Pull request by darranl**
This is a resubmission of https://github.com/wildfly-security/wildfly-elytron/pull/1899 to address the merge conflict.

https://issues.redhat.com/browse/ELY-2548

@Skyllarr FYI you had approved the original submission of this PR.

### CODE REVIEW

File: tests/base/src/test/java/org/wildfly/security/http/impl/AbstractBaseHttpTest.java [hunk 1]
```diff
@@ -483,6 +483,11 @@ protected CallbackHandler getCallbackHandler(String username, String realm, Stri
                 } else if (callback instanceof AuthorizeCallback) {
                     if (token != null) {
                         ((AuthorizeCallback) callback).setAuthorized(true);
+                    } else if(username.equalsIgnoreCase("unauthorizedUser")){
+                        ((AuthorizeCallback) callback).setAuthorized(false);
+                    } else if(username.equals(((AuthorizeCallback) callback).getAuthenticationID()) &&
+                       username.equals(((AuthorizeCallback) callback).getAuthorizationID())) {
+                        ((AuthorizeCallback) callback).setAuthorized(true);
                     } else {
                         if (username.equals(((AuthorizeCallback) callback).getAuthenticationID()) &&
```
Comment (skyllarr) [inline tests/base/src/test/java/org/wildfly/security/http/impl/AbstractBaseHttpTest.java:492]: @darranl This is a minor but these conditions can be simplified, we can merge this now and then I can create a good-first-issue to simplify this

**Review by skyllarr — APPROVED** [review]
@darranl Thank you!

File: tests/base/src/test/java/org/wildfly/security/http/impl/AbstractBaseHttpTest.java (hunk 1 above)
Comment (darranl) [reply tests/base/src/test/java/org/wildfly/security/http/impl/AbstractBaseHttpTest.java:492]: IMO I would say go ahead and raise a good first issue - this is not my fix I just addressed a merge conflict so we can clear some of the PR queue backlog.

**Comment by darranl** [comment]
Adding my approval (for the original PR as this PR is just to fix a merge conflict)

**Pull request merged**